apps/controllers/midibox_ng_v1/src/mbng_event.c

```diff
@@ -4089,6 +4222,33 @@ s32 MBNG_EVENT_ItemReceive(mbng_event_item_t *item, u16 value, u8 from_midi, u8
 	}
 	break;
 
+	  //Mackie C4 Control special case
+	  //considers different encoder rotation accelerations
+      case MBNG_EVENT_ENC_MODE_C4ENC:
+    	  if( item->value != 0x00 ) {
+    		  if(item->value < 0x40){
+    			  switch(item->value){

```

Comment:
Instead of a switch() statement it would be better to use range checks here in case values are processed which are not explicitly mentioned here (for whatever reason…)

E.g.:
if( item->value <= 1 ) incrementer = 1; else if( item->value <=4 ) incrementer = 2;
else if( item->value <=8 ) incrementer= 4;
…